test-case/check-runtime-pm-double-active.sh

```diff
@@ -0,0 +1,145 @@
+func_check_dsp_status()
+{
+    # more frequent checking of the runtime pm status is required. Every loop is 100ms
+    local iloop=$(expr 10 '*' "$1")
+    dlogi "wait dsp power status to become suspended, starting max [100ms x $iloop times] checking"
+    for i in $(seq 1 $iloop)
+    do
+        # Here we pass a hardcoded 0 to python script, and need to ensure
+        # DSP is the first audio pci device in 'lspci', this is true unless
+        # we have a third-party pci sound card installed.
+        [[ $(sof-dump-status.py --dsp_status 0) == "suspended" ]] && break
+        sleep 0.1
+        if [ $i -eq $iloop ]; then
+            dlogi "dsp is not suspended after $1s, end test"
+            exit 1
+        fi
+    done
+    dlogi "dsp suspended in $i try, ${i}00 msec"
+}
+
+func_opt_parse_option $*
+tplg=${OPT_VALUE_lst['t']}
+loop_count=${OPT_VALUE_lst['l']}
+[[ -z $tplg ]] && dloge "Miss tplg file to run" && exit 1
+
+[[ $(sof-dump-status.py --dsp_status 0) == "unsupported" ]] &&
+    dlogi "platform doesn't support runtime pm, skip test case" && exit 2
+
+declare -A APP_LST DEV_LST
+APP_LST['playback']='aplay'
+DEV_LST['playback']='/dev/zero'
+APP_LST['capture']='arecord'
+DEV_LST['capture']='/dev/null'
+
+[[ ${OPT_VALUE_lst['s']} -eq 1 ]] && func_lib_start_log_collect
+func_pipeline_export $tplg
+func_lib_setup_kernel_last_line
+
+for idx in $(seq 0 $(expr $PIPELINE_COUNT - 1))
+do
+    channel=$(func_pipeline_parse_value $idx channel)
+    rate=$(func_pipeline_parse_value $idx rate)
+    fmt=$(func_pipeline_parse_value $idx fmt)
+    dev=$(func_pipeline_parse_value $idx dev)
+    pcm=$(func_pipeline_parse_value $idx pcm)
+    type=$(func_pipeline_parse_value $idx type)
+    snd=$(func_pipeline_parse_value $idx snd)
+
+    cmd="${APP_LST[$type]}"
+    dummy_file="${DEV_LST[$type]}"
+    [[ -z $cmd ]] && dloge "$type is not supported, $cmd, $dummy_file" && exit 1
+
+    for i in $(seq 1 $loop_count)
+    do
+        dlogi "Iteration $i of $loop_count for $pcm"
+        # playback or capture device - check status
+        dlogc "$cmd -D $dev -r $rate -c $channel -f $fmt $dummy_file -q"
+        $cmd -D $dev -r $rate -c $channel -f $fmt $dummy_file -q &
+        pid=$!
+
+        # TODO: delay 2.5s is workaround for the SSH aplay delay issue.
+        sleep 2.5
+
+        kill -0 $pid
+        if [ $? -ne 0 ]; then
+            func_lib_lsof_error_dump $snd
+            dloge "$cmd process for pcm $pcm is not alive"
+            exit 1
+        fi
+
+        [[ -d /proc/$pid ]] && result=`sof-dump-status.py --dsp_status 0`
+
+        dlogi "runtime status: $result"
+        if [[ $result == active ]]; then
+            # stop playback or capture device - check status again
+            dlogc "kill process: kill -9 $pid"
+            kill -9 $pid && wait $pid 2>/dev/null
+            dlogi "$cmd killed"
+            func_check_dsp_status ${OPT_VALUE_lst['d']}
+            result=`sof-dump-status.py --dsp_status 0`
```

Comment:
`func_check_dsp_status` has already make sure status is "suspend", otherwise it will exit. So the below `if` will never happen.